design-documents/graph-ql/coverage/customer-orders.md

```diff
@@ -0,0 +1,210 @@
+}
+
+type RefundChildProduct implements ChildOrderProductInterface{
+    refunded: Float! # number of refunded items
+}
+```
+
+### Payment Method Schema
+
+To provide more customization for different payment solutions, the payment method will be represented by own type instead of simple string:
+
+```graphql
+#  payment method used to pay for the order
+type PaymentMethod {
+    name: String! # payment method name for e.g Braintree, Authorize etc
+    type: String! # payment method type used to pay for the order for e.g Credit Card, PayPal etc.
+    additonal_data: [KeyValue] # additional data per payment method type
+}
+```
+
+### Pricing Schema
+
+As entities like order, invoice, credit memo might have complex pricing type:
+
+```graphql
+interface OrderPricingInterface {
+    subtotal: Float! # order subtotal excluding, shipping, discounts and tax
+    discounts: [Discount] # all the discounts applied to the order
+    shipping_handling: Float! # shipping and hnadling for the order
+    tax: Float! # tax applied to the order
+    grand_total: Float! # final order total including shipping and taxes
+}
+​
+type OrderPricing implements OrderPricingInterface {
+​
+}
+```
+
+## Invoice Type Schema
+
+The invoice entity will have the similar to the order schema:
+
+```graphql
+type Invoice {
+    number: String! # user friendly identifier for the invoice
+    pricing: OrderPricingInterface! # invoice pricing details
+    products: [OrderProductInterface]! # invoiced product details
+}
+
+type InvoiceProduct implements OrderProductInterface{
+    invoiced: Float! # number of invoided items
+}
+
+type InvoiceChildProduct implements ChildOrderProductInterface{
+    invoiced: Float! # number of invoided items
+}
+
+type InvoicePricing implements OrderPricingInterface {
+  
+}
+```
+
+## Refund Type Schema
+
+The credit memo entity will have the similar to the order and invoice schema:
+
+```graphql
+type Refund {
+    number: String!
+    items: [OrderProductInterface]! # items refunded
+    pricing: OrderPricingInterface! # refund pricing details
+}
+
+type RefundProduct implements OrderProductInterface{
+    refunded: Float! # number of refunded items
+}
+
+type RefundPricing implements OrderPricingInterface {
+    refunded_to_payment_method: Float # amount refunded
+}
+```
+
+## Shipment Type Schema
+
+```graphql
+type OrderShipment {
+    shipping_method: String! # shipping method for the order
+    shipping_address: CustomerAddress! # shipping address for the order
+    tracking_link: String # tracking link for the order
+    shipped_items: [OrderProductInterface] # items included in the shipping
+}
+ 
+type ShipmentProduct implements OrderProductInterface{
+    shipped: Float! #number of shipped items
+}
+
+type ShipmentChildProduct implements ChildOrderProductInterface{
+    shipped: Float! # umber of shipped items
+}
+```
+
+## Additional Types
+
+The `NameValue` type will provide a possibility to use key-value pairs:
```

Comment:
Is this supposed to be KeyValue, or are fields currently using KeyValue actually supposed to be NameValue?